SampleAPI/Repository/EmployeeRepository.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SampleAPI.Constant;
+using SampleAPI.Contexts;
+using SampleAPI.Models;
+
+namespace SampleAPI.Repository
+{
+    /// <summary>
+    /// EmployeeRepository is used to perform CRUD operations with employee entity.
+    /// </summary>
+    public class EmployeeRepository : IEmployeeRepository
+    {
+        private readonly EmployeeContext _dbContext;
+
+        public EmployeeRepository(EmployeeContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        /// <summary>
+        /// Add new employee record to the database.
+        /// </summary>
+        /// <param name="employeeModel">Employee model</param>
+        public async Task AddAsync(EmployeeModel employeeModel)
+        {
+            if (employeeModel != null)
+            {
+                _dbContext.Employees.Add(employeeModel);
+                await _dbContext.SaveChangesAsync();
+            }
+        }
+
+        /// <summary>
+        /// Delete employee from the database.
+        /// </summary>
+        /// <param name="employeeId">Employee model.</param>
+        public async Task DeleteAsync(int employeeId)
+        {
+            var employee = await _dbContext.Employees.FindAsync(employeeId);
+
+            if (employee == null)
+            {
+                throw new CustomException(AppConstants.NoRecordErrorMessage);
+            }
+
+            _dbContext.Employees.Remove(employee);
+            await _dbContext.SaveChangesAsync();
+        }
+
+        /// <summary>
+        /// Get the employee list.
+        /// </summary>
+        /// <returns>Employee list.</returns>
+        public async Task<IEnumerable<EmployeeModel>> GetAllAsync(string searchText, string role, bool includeInActive)
+        {
+             var employeeList = await _dbContext.Employees.ToListAsync();
```

Comment:
we would go for pagination when the data is huge, but here we are first getting all of the data.

Reply:
this could be 
`_dbContext.Employees
.Where( x =>
   (string.IsNullOrWhiteSpace(searchText) || x.Name.Contains(searchText) || x.Role.Contains(searchText)) &&
   (string.IsNullOrWhiteSpace(role) || x.Role.Equals(role)) &&
   (includeInActive ||  x.IsActive)
)`